### PR TITLE
Application Architecture - MVVM

### PR DESCRIPTION
**Pull request by Dynna**

### CODE REVIEW

File: MyApplication/app/src/main/java/com/example/myapplication/ui/main/view/UserActivity.kt [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+                    Status.ERROR -> {
+                        recyclerView.visibility = View.VISIBLE
+                        progressBar.visibility = View.GONE
+                        Toast.makeText(this, it.message, Toast.LENGTH_LONG).show()
+                    }
+                    Status.LOADING -> {
+                        progressBar.visibility = View.VISIBLE
+                        recyclerView.visibility = View.GONE
+                    }
+                }
```
Comment (ASV44) [inline MyApplication/app/src/main/java/com/example/myapplication/ui/main/view/UserActivity.kt:71]: This is business logic of your app. It handles different states. You should be able to test it. That's why you have to move it somewhere in that way to be able to cover it in unit tests

File: MyApplication/app/src/main/java/com/example/myapplication/ui/main/viewmodel/MainViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+    fun getUsers() = liveData(Dispatchers.IO) {
+        emit(Resource.loading(data = null))
+        try {
+            emit(Resource.success(data = mainRepository.getUsers()))
+        } catch (exception: Exception) {
+            emit(Resource.error(data = null, message = exception.message ?: "Error Occurred!"))
+        }
+    }
+}
```
Comment (ASV44) [inline MyApplication/app/src/main/java/com/example/myapplication/ui/main/viewmodel/MainViewModel.kt:20]: @Dynna familiarize yourself with `ViewModel` testing before starting lab5. For [example](https://blog.mindorks.com/unit-testing-viewmodel-with-kotlin-coroutines-and-livedata)

**Review by ASV44 — CHANGES_REQUESTED** [review]

**Review by ASV44 — CHANGES_REQUESTED** [review]
Please also rebase this branch in top of main

File: MyApplication/app/src/main/java/com/example/myapplication/ui/main/view/UserActivity.kt (hunk 1 above)
Comment (Dynna) [reply MyApplication/app/src/main/java/com/example/myapplication/ui/main/view/UserActivity.kt:71]: I see no problem in using this logic here as I already call getUsers() function from ViewModel where business logic actually is.
The example I followed: https://stackoverflow.com/questions/57966523/where-to-put-business-logic-when-working-with-mvvm

**Comment by Dynna** [comment]
> Please also rebase this branch in top of main

done

**Review by ASV44 — APPROVED** [review]
@Dynna PR approved. Feel free to merge it

**Pull request merged**